ext/drreg/drreg.c

```diff
@@ -104,14 +117,26 @@ typedef struct _reg_info_t {
 #define DR_REG_EFLAGS DR_REG_INVALID
 
 #define GPR_IDX(reg) ((reg)-DR_REG_START_GPR)
+#define SIMD_IDX(reg) ((reg_resize_to_opsz(reg, OPSZ_64)) - DR_REG_START_ZMM)
+
+/* Depending on architecture, we have a set of applicable SIMD registers */
+#define DR_REG_APPLICABLE_START_SIMD DR_REG_START_ZMM
+#define DR_REG_APPLICABLE_STOP_SIMD \
+    (DR_REG_APPLICABLE_START_SIMD + MCXT_NUM_SIMD_SLOTS - 1)
```

Comment:
We are reasoning over ZMM registers, BUT we are not considering entire ZMM ranges, only those within the MCXT_NUM_SIMD_SLOTS bound depending on the architecture. On 32-bit, there are only 8 XMM registers.

Reply:
I see. We have proc_num_simd_saved() for that. Problem here: it can switch dynamically, lazily from avx size to avx-512 size. So unless we do sth dynamic as mentioned [here](https://github.com/DynamoRIO/dynamorio/pull/3893#discussion_r335233429), probably statically assuming the max is best.